drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi_dma.c

```diff
@@ -309,13 +302,10 @@ static int transceive_dma(const struct device *dev, const struct spi_config *spi
 		return ret;
 	}
 
-#ifdef CONFIG_SOC_SERIES_MCXN
-	base->TCR |= LPSPI_TCR_CONT_MASK;
-#endif
-
-	/* DMA is fast enough watermarks are not required */
-	LPSPI_SetFifoWatermarks(base, 0U, 0U);
-
+	/* Set a reasonable watermark level to avoid transfer pause because of
+	 * TX FIFO empty. Half of FIFO size is always good for DMA mode.
+	 */
+	LPSPI_SetFifoWatermarks(base, config->tx_fifo_size/2, 0U);
```

Comment:
I understand the fix you're trying to do, and RM does say that the DMA request can be generated by the fifo watermark signals, so it's a good idea, but I am concerned that the spi context might now be getting updated wrong, because right now this driver is (naively)  written so that it expects both the RX and TX will be the same size and updates the context with that size, assuming they both finished. I guess if this is true then the RX will finish when the TX also finished? And DMA will preload some data before then, so maybe it works, but maybe comment about this actual behavior going on in the file, since it's maybe not going to be immediately obvious to everyone who reads this file how it is actually working.